api/apps/api/src/modules/scenarios/scenarios.service.ts

```diff
@@ -548,4 +553,17 @@ export class ScenariosService {
     }
     return right(true);
   }
+
+  private async canEditGuard(scenarioId: string) {
+    const scenario = await this.crudService.getById(scenarioId);
+    // TODO: Check where do we should use the guard
+    const editIsBlocked = await this.projectChecker.hasProjectPendingExports(
+      scenario.projectId,
+    );
+
+    if (isLeft(editIsBlocked))
+      throw new BadRequestException(
+        `Scenario ${scenarioId} editing is blocked because of pending export`,
+      );
+  }
```

Comment:
this is fine for the time being - I think at some point we'll need to move it to its own module (maybe even a project library) if we want to reuse the same logic across all the spots in the codebase where we may want to reject starting a change if there is an export pending on the parent project.

Reply:
Mmm maybe some kind of provider that we could inject in the services that need it could be enough.
Something like:
```typescript 
export class ProjectEditGuardRealImplementationWithABetterName extends ProjectEditGuard {}
......
// And then we could use it like this in just one line:
await this.projectEditGuard.ensureEdittingIsAllowed(scenarioId)
```
If you like this (I actually do, didn't think about this before) we can implement it quickly!

Reply:
yep, I agree with this approach.

I'd recommend to keep it as is in this PR and move it to a provider in a separate task (but up to you)

Reply:
I think that it would be ok to abstract it in this PR to kept it clean 🤔